src/azure-cli/azure/cli/command_modules/vm/azure_stack/commands.py

```diff
@@ -0,0 +1,603 @@
+
+    with self.command_group('ppg', compute_proximity_placement_groups_sdk, min_api='2018-04-01', client_factory=cf_proximity_placement_groups) as g:
+        g.custom_command('create', 'create_proximity_placement_group', validator=process_ppg_create_namespace)
+        g.custom_command('list', 'list_proximity_placement_groups')
+        g.generic_update_command('update', setter_name='create_or_update', custom_func_name='update_ppg')
+
+    with self.command_group('vm monitor log', client_factory=cf_log_analytics_data_plane) as g:
+        g.custom_command('show', 'execute_query_for_vm', transform=transform_log_analytics_query_output)  # pylint: disable=show-command
+
+    with self.command_group('vm monitor metrics', custom_command_type=monitor_custom, command_type=metric_definitions_sdk, resource_type=ResourceType.MGMT_MONITOR, operation_group='metric_definitions', min_api='2018-01-01', is_preview=True) as g:
+        from azure.cli.command_modules.monitor.transformers import metrics_table, metrics_definitions_table
+        from azure.cli.core.profiles._shared import APIVersionException
+        try:
+            g.custom_command('tail', 'list_metrics', command_type=monitor_custom, table_transformer=metrics_table)
+            g.command('list-definitions', 'list', table_transformer=metrics_definitions_table)
+        except APIVersionException:
+            pass
+
+    with self.command_group('capacity reservation group', capacity_reservation_groups_sdk, min_api='2021-04-01',
+                            client_factory=cf_capacity_reservation_groups) as g:
+        g.custom_command('create', 'create_capacity_reservation_group')
+        g.custom_command('update', 'update_capacity_reservation_group')
+        g.custom_show_command('show', 'show_capacity_reservation_group')
+
+    with self.command_group('capacity reservation', capacity_reservations_sdk, min_api='2021-04-01',
+                            client_factory=cf_capacity_reservations) as g:
+        g.custom_command('create', 'create_capacity_reservation', supports_no_wait=True)
+        g.custom_command('update', 'update_capacity_reservation', supports_no_wait=True)
+        g.custom_show_command('show', 'show_capacity_reservation')
+
+    with self.command_group('restore-point', restore_point, client_factory=cf_restore_point, min_api='2021-03-01') as g:
+        g.custom_show_command('show', 'restore_point_show')
+        g.custom_command('create', 'restore_point_create', supports_no_wait=True)
+        g.wait_command('wait')
+
+    with self.command_group('restore-point collection', restore_point_collection, min_api='2021-03-01',
+                            client_factory=cf_restore_point_collection) as g:
+        g.custom_show_command('show', 'restore_point_collection_show')
+        g.custom_command('create', 'restore_point_collection_create')
+        g.custom_command('update', 'restore_point_collection_update')
+        g.wait_command('wait')
```

Comment:
> min_api='2021-03-01'

Please remove these code unrelated to Azure Stack in future PRs

Reply:
will remove them in a future migration PR

Reply:
OK, sounds good